packages/amplication-server/src/core/moduleDto/moduleDto.service.ts

```diff
@@ -137,8 +157,34 @@ export class ModuleDtoService extends BlockTypeService<
         "AND"
       );
     } else {
-      return [];
+      allDTOs = [];
     }
+
+    const resourceId = args.where.resource?.id;
+    return this.filterDtosByGeneratorName(resourceId, allDTOs);
+  }
+
+  async filterDtosByGeneratorName(
+    resourceId: string,
+    allDTOs: ModuleDto[]
+  ): Promise<ModuleDto[]> {
+    const resource = await this.resourceService.findOne({
+      where: { id: resourceId },
+    });
+
+    if (
+      !resource ||
+      (resource &&
+        CODE_GENERATOR_NAME_TO_ENUM[resource.codeGeneratorName] !==
+          EnumCodeGenerator.DotNet)
+    )
+      return allDTOs;
+
+    return allDTOs.filter((dto) => {
```

Comment:
@mulygottlieb, you aren't using the results, and it looks like you don't override the original array

Reply:
what do you mean? I return the result from the .filter() call so the return value is the filtered list